src/sync/all.rs

```diff
@@ -2218,6 +2218,7 @@ impl<TRq> Shared<TRq> {
             max_disjoint_headers: self.max_disjoint_headers,
             max_requests_per_block: self.max_requests_per_block,
             full: false,
+            banned_blocks: iter::empty(), // TODO: not implemented, should be passed by config after the optimistic sync supports banned blocks too
```

Comment:
That's a good question.
I've been using TODOs throughout the code to track what is missing, because maintaining issues for code that is still changing is usually not great.
My general idea is: if the code is in a clean state, then we should use issues to track its defects, but if it's code that still needs a lot of changes then writing these changes in TODOs is more appropriate. But I'm open to opinions.

Reply:
TODOs are fine! My proposal was more about adding issues along with todos and adding a link to an issue to the TODO. The problem with TODOs is you don't know if somebody's is working on them right now. But that may be not the problem since there are only two of us and you know more or less what I'm working on.